front/app/api/navbar/useUpdateNavbarItem.test.ts

```diff
@@ -0,0 +1,57 @@
+import { renderHook, act } from '@testing-library/react-hooks';
+
+import useUpdateNavbarItem from './useUpdateNavbarItem';
+import { navbarItemsData } from './__mocks__/useNavbarItems';
+
+import { setupServer } from 'msw/node';
+import { rest } from 'msw';
+
+import createQueryClientWrapper from 'utils/testUtils/queryClientWrapper';
+
+const apiPath = '*nav_bar_items/:id';
+const server = setupServer(
+  rest.patch(apiPath, (_req, res, ctx) => {
+    return res(ctx.status(200), ctx.json({ data: navbarItemsData[0] }));
+  })
+);
+
+describe('useUpdateNavbarItem', () => {
+  beforeAll(() => server.listen());
+  afterAll(() => server.close());
+
+  it('mutates data correctly', async () => {
+    const { result, waitFor } = renderHook(() => useUpdateNavbarItem(), {
+      wrapper: createQueryClientWrapper(),
+    });
+
+    act(() => {
+      result.current.mutate({
+        id: 'id',
+        title_multiloc: { en: 'name' },
+      });
+    });
+
+    await waitFor(() => expect(result.current.isSuccess).toBe(true));
+    expect(result.current.data?.data).toEqual(navbarItemsData[0]);
+  });
```

Comment:
I don't fully get these tests at the moment. So we're mutating result.current with certain data, but the end result should equal our imported navbarItemsData, not the data we passed?

Reply:
Yes, this was a mistake on my end. Indeed, we should expect the full list.

Reply:
We are not making an actual api request here, just mocking it. So it's testing that what is returned from the fetcher is what is returned by the hook afterwards and we define both to be equal in the test and then make an assertion on that basis.